src/View/BookingCodes.php

```diff
@@ -76,4 +430,66 @@ public static function renderCSV( $timeframeId = null ) {
 		die;
 	}
 
+    /**
+     * action for sending Booking Codes by E-mail
+     * 
+     * @param int $timeframeId      ID of requested timeframe
+     * @param int $tsFrom           Timestamp of first Booking Code
+     * @param int $tsTo             Timestamp of last Booking Code
+     */
+    public static function emailCodes($timeframeId = null, $tsFrom=null, $tsTo=null) { 
+ 
+        if($timeframeId == null) {
+            $timeframeId = sanitize_text_field(@$_GET['post']);
+        }
+        if(empty($timeframeId)) {
+            wp_die(
+                commonsbooking_sanitizeHTML( __( "Unable to retrieve Booking Codes", "commonsbooking" ) ),
+                commonsbooking_sanitizeHTML( __( "Missing ID of timeframe post", "commonsbooking" ) )
+            );
+        }
+
+        if($tsFrom == null) {
+            $tsFrom = absint(@$_GET['from']);
+        }
+        if($tsTo == null) {
+            $tsTo = absint(@$_GET['to']);
+        }
+
+
+        add_action( 'commonsbooking_mail_sent',function($action,$result){
+            if(is_wp_error($result))
+            {
+                wp_die($result);
+            }
+            elseif($result === false)
+            {
+                wp_die(
+                    commonsbooking_sanitizeHTML( __( "Unable to send email", "commonsbooking" ) ),
```

Comment:
Ich würde hier lieber einen Error transient nutzen, der ist schon im Einsatz beim Model\Timeframe in der isValid() Funktion, dann muss dafür keine neue Seite geöffnet werden aber die Nutzenden kriegen trotzdem (etwas eleganter) die Fehlermeldung. Würde dafür aber vielleicht einen neuen Fehlertypen einrichten, der muss dann nur in der Plugin.php in der renderError Funktion ergänzt werden

Reply:
Das würde auf jeden Fall benutzerfreundlicher sein aber das würde ja nicht nur für die Fehlermeldung sondern auch für die Erfolgsmeldung gelten.
D. h. es müsste dann in der Plugin.php auch noch eine renderSuccess Methode implementiert werden (ich habe zumindest keine gefunden). Fände ich gut, wollte nur nicht zuviel in die Core Funktionen eingreifen.